src/simulation.jl

```diff
@@ -680,15 +677,16 @@ end
 Base.showerror(io::IO, e::UnsupportedStateeltypeException) = print(io, "Decapodes does not support state element types as $(e.type), only Float32 or Float64")
 
 """
-    gensim(user_d::SummationDecapode, input_vars::Vector{Symbol}; dimension::Int=2, stateeltype::DataType = Float64, code_target::GenerationTarget = CPUTarget())
+    gensim(user_d::SummationDecapode, input_vars::Vector{Symbol}; dimension::Int=2, stateeltype::DataType = Float64, code_target::AbstractGenerationTarget = CPUTarget(), can_prealloc::Bool = true)
 
 Generates the entire code body for the simulation function. `user_d` is the user passed Decapodes which will be left unmodified and 'input_vars' is the collection of
 state variables and literals in the Decapode.
 
 Optional keyword arguments are `dimension`, which is the dimension of the problem and defaults to 2D, `stateeltype`, which is the element type of the state forms and
-defaults to Float64 and `code_target`, which is the intended architecture target for the generated code, defaulting to regular CPU compatible code.
+defaults to Float64, `code_target`, which is the intended architecture target for the generated code, defaulting to regular CPU compatible code, and 'can_prealloc' which
+is set to `true` by default and determines if intermediate results can be preallocated.
```

Comment:
Is setting this flag necessary for using implicit solvers? And if so can that be documented here?

Reply:
No, but if you want to find a sparse Jacobian pattern, plus some other more involved implicit solver things I forget exactly what, then you do need this on.

Reply:
But this is more so a "safe" mode, where there's less optimization stuff going on that could impact solving, such as needing to find a sparse Jacobian.

Reply:
Can you document that either in this docstring or in a blurb in the Docs page

Reply:
Yeah, I can do that when I make a docs page on Brusselator.

Reply:
Ok. For the time being can you put something in this doctoring then.

Reply:
Done and I've also cleaned up the whole docstring since this'll be important to the user.

Reply:
I mean can you mention in the docstring the "why" of this flag (Get JacSparsity to work etc.). The "what" is excellently answered.